From 9ebb51d8853ffab5dfac8b8b6035500c13ecf385 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 10 Dec 2022 21:39:02 +0000
Subject: [PATCH 5/6] gnu: python-astropy: Update to 5.1.1.

* gnu/packages/astronomy.scm (python-astropy): Update to 5.1.1.
  [source]{snippet}: Unbundle wcslib.
  [arguments]{phases}: preparations: Use all system libraries.
  [inputs]: Add wcslib.
---
 gnu/packages/astronomy.scm | 20 ++++++++------------
 1 file changed, 8 insertions(+), 12 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 1876b0cde0..030ae9b4ea 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -982,25 +982,23 @@ (define-public celestia-gtk
 (define-public python-astropy
   (package
     (name "python-astropy")
-    (version "5.1")
+    (version "5.1.1")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "astropy" version))
        (sha256
-        (base32 "0zkv3ddzlxx21i796azfbqxrqnsxn83vsczscv577iyzxp3v5c8x"))
+        (base32 "10dxjim60ch4qznqa5s63q936mkvy95p0k26kcwzv43hmybdcjxs"))
        (modules '((guix build utils)))
        (snippet
         '(begin
            ;; Remove Python bundles.
            (with-directory-excursion "astropy/extern"
              (for-each delete-file-recursively '("ply" "configobj")))
-           ;; Remove cextern bundles and leave the wcslib bundle.  Astropy
-           ;; upgrades to different versions of wcslib every few releases
-           ;; and tests break every upgrade.
-           ;; TODO: unbundle wcslib.
+           ;; Remove cextern bundles. Check bundled versions against available
+           ;; in Guix in the future update of astropy.
            (with-directory-excursion "cextern"
-             (for-each delete-file-recursively '("cfitsio" "expat")))
+             (for-each delete-file-recursively '("cfitsio" "expat" "wcslib")))
            #t))))
     (build-system python-build-system)
     (arguments
@@ -1008,10 +1006,8 @@ (define-public python-astropy
        (modify-phases %standard-phases
          (add-after 'unpack 'preparations
            (lambda _
-             ;; Use our own libraries in place of bundles, with the
-             ;; exception of wcslib.
-             (setenv "ASTROPY_USE_SYSTEM_CFITSIO" "1")
-             (setenv "ASTROPY_USE_SYSTEM_EXPAT" "1")
+             ;; Use our own libraries in place of bundles.
+             (setenv "ASTROPY_USE_SYSTEM_ALL" "1")
              ;; Some tests require a writable home.
              (setenv "HOME" "/tmp")
              ;; Relax xfail tests.
@@ -1058,7 +1054,7 @@ (define-public python-astropy
            python-sgp4
            python-skyfield))
     (inputs
-     (list cfitsio expat))
+     (list cfitsio expat wcslib))
     (propagated-inputs
      (list python-configobj
            python-numpy
-- 
2.38.1

